kodi_addon_checker/pylint_checks.py

```diff
@@ -0,0 +1,31 @@
+import io
+import json
+import os
+
+from pylint.lint import Run
+from terminaltables import AsciiTable
+from pylint.reporters.json import JSONReporter
+
+from .report import Report
+from .record import Record, WARNING
+
+# TODO: Do not repeat name of same file in report just do something like . utils.py - 90,1780 ?
+table_format = [["File path", "Line no", "Message"]]
+
+
+def short_path(path):
+    return os.path.split(path)[1]
+
+
+def analyze(report: Report, filename: str):
+
+    ARGS = ["-r", "n", "--score=yes", "--lint-all=y", "--rcfile=kodi_addon_checker/.pylintrc"]
+    out = io.StringIO()
+    Run([filename] + ARGS, reporter=JSONReporter(out), do_exit=False)
+
+    json_data = json.loads(out.getvalue())
+    for issue in json_data:
+        table_format.append([short_path(issue['path']), issue['line'], issue['message']])
```

Comment:
Can we extract the issue code number (e.g `E0601`) from the issue? It might be a good idea to prepare the checker for some post-processing on the collected issues before returning. Example, pylint will report `import routing` as a wrong import as the module can't be found in site-packages but the dev might be importing it as a script module in addon.xml. This will lead to a lot of false positives.